src/components/LayerCollection/LayerPO.js

```diff
@@ -25,16 +27,11 @@ class LayerPO extends Component {
     this.connectedReceived = props.stripes.connect(Received);
   }
 
-  updatePO(data) {
-    const deep = cloneDeep(data);
+  updatePO(order) {
+    const { parentMutator, onCancel } = this.props;
 
-    delete deep.created_by_name;
-    delete deep.assigned_to_user;
-    delete deep.vendor_name;
-    delete deep.po_lines;
-    this.props.parentMutator.records.PUT(deep).then(() => {
-      this.props.onCancel();
-    });
+    updateOrderResource(order, parentMutator.records)
```

Comment:
try to make this function `async` and `await updateOrderResource(order, parentMutator.records)`